### PR TITLE
Adding Commandlets for Cached Structural Navigation

### PR DESCRIPTION
**Pull request by ParamveerSisodia**
These commandlets allow tenant admins to enable/disable caching for webs where structural navigation is enabled to boost page load performance

### CODE REVIEW

**Comment by get-itips** [comment]
Hello @ParamveerSisodia 
Please add missing metadata to the documents

an example

```
applicable: SharePoint Online
title: Get-SPOSite
schema: 2.0.0
author: trent-green
ms.author: trgreen
ms.reviewer:
---
```

and the cmdlets in Related Links should be Links to the those cmdlet's articles.

Let us know if you need assistance with markdown formatting.

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOStructuralNavigationCacheSiteState.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+Get the structural navigation caching state for a site collection.
+
+## SYNTAX
+
+```
+Get-SPOStructuralNavigationCacheSiteState -SiteUrl <String> [<CommonParameters>]
+```
+
+## DESCRIPTION
+The Get-SPOStructuralNavigationCacheSiteState cmdlet can be used to determine if structural navigation caching is enabled or disabled for a site collection. [Learn more](https://support.office.com/en-us/article/structural-navigation-and-performance-f163053f-8eca-4b9c-b973-36b395093b43).
```
Comment (get-itips) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPOStructuralNavigationCacheSiteState.md:20]:
```suggestion
The Get-SPOStructuralNavigationCacheSiteState cmdlet can be used to determine if structural navigation caching is enabled or disabled for a site collection. [Learn more](https://support.office.com/article/structural-navigation-and-performance-f163053f-8eca-4b9c-b973-36b395093b43). 

```

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOStructuralNavigationCacheSiteState.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    For more information, type: "get-help Get-SPOStructuralNavigationCacheSiteState -detailed". 
+
+    For technical information, type: "get-help Get-SPOStructuralNavigationCacheSiteState -full". 
+
+    For online help, type: "get-help Get-SPOStructuralNavigationCacheSiteState -online" 
+
+ 
+## RELATED LINKS
+Get-SPOStructuralNavigationCacheWebState
```
Comment (get-itips) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPOStructuralNavigationCacheSiteState.md:69]:
```suggestion
[Get-SPOStructuralNavigationCacheWebState](Get-SPOStructuralNavigationCacheWebState.md) 
```

File: sharepoint/sharepoint-ps/sharepoint-online/Get-SPOStructuralNavigationCacheWebState.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+Get the structural navigation caching state for a web. 
+
+## SYNTAX
+
+```
+Get-SPOStructuralNavigationCacheWebState -WebUrl <String> [<CommonParameters>]
+```
+
+## DESCRIPTION
+The Get-SPOStructuralNavigationCacheWebState cmdlet can be used to determine if structural navigation caching is enabled or disabled for a web in a site collection. [Learn more](https://support.office.com/en-us/article/structural-navigation-and-performance-f163053f-8eca-4b9c-b973-36b395093b43).
```
Comment (tseward) [inline sharepoint/sharepoint-ps/sharepoint-online/Get-SPOStructuralNavigationCacheWebState.md:20]: Please remove `en-us` from the link.

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOStructuralNavigationCacheSiteState.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+Enable or disable caching for all webs in a site collection. 
+
+## SYNTAX
+
+```
+Set-SPOStructuralNavigationCacheSiteState -SiteUrl <String> -IsEnabled <Boolean> [<CommonParameters>]
+```
+
+## DESCRIPTION
+The Set-SPOStructuralNavigationCacheSiteState cmdlet can be used to enable or disable caching for all webs in a site collection.[Learn more](https://support.office.com/en-us/article/structural-navigation-and-performance-f163053f-8eca-4b9c-b973-36b395093b43).
```
Comment (tseward) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOStructuralNavigationCacheSiteState.md:20]: Please remove `en-us` from the link

File: sharepoint/sharepoint-ps/sharepoint-online/Set-SPOStructuralNavigationCacheWebState.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+Enable or disable caching for a web in a site collection. 
+
+## SYNTAX
+
+```
+Set-SPOStructuralNavigationCacheWebState -WebUrl <String> -IsEnabled <Boolean> [<CommonParameters>]
+```
+
+## DESCRIPTION
+The Set-SPOStructuralNavigationCacheWebState cmdlet can be used to enable or disable caching for a web in a site collection.[Learn more](https://support.office.com/en-us/article/structural-navigation-and-performance-f163053f-8eca-4b9c-b973-36b395093b43).
```
Comment (tseward) [inline sharepoint/sharepoint-ps/sharepoint-online/Set-SPOStructuralNavigationCacheWebState.md:20]: Please remove `en-us` from the link

**Comment by kenwith** [comment]
@ParamveerSisodia - looks like there are a number of issues with the PR and also each one is error on the automatic schema checks. Detailed guidance on what needs to be in the references can be found here: https://github.com/MicrosoftDocs/office-docs-powershell/blob/master/repo_docs/NEW_CMDLETS.md

@get-itips has created new cmdlet references for other teams and it might be easier to work with him to have him do it for you. If you want to do that let me know and I will close this one.

**Comment by ParamveerSisodia** [comment]
@kenwith , @tseward , @dariomws   and @get-itips  i have update the PR. Kindly review and let me know if i missed something

File: sharepoint/sharepoint-ps/sharepoint-online/sharepoint-online.md [hunk 6]
```diff
@@ -191,6 +191,14 @@ The following cmdlet references are for SharePoint Online.
 
 {{Manually Enter Get-SPOSiteScript Description Here}}
 
+### [Get-SPOStructuralNavigationCacheSiteState](Get-SPOStructuralNavigationCacheSiteState.md)
+
+{{Manually Enter Get-SPOSiteScript Description Here}}
```
Comment (get-itips) [inline sharepoint/sharepoint-ps/sharepoint-online/sharepoint-online.md:196]:
```suggestion
{{Manually Enter Get-SPOStructuralNavigationCacheSiteState Description Here}}
```

**Comment by ParamveerSisodia** [comment]
@get-itips updated as per your suggestions

**Pull request merged**